lib/common/strings.c

```diff
@@ -782,3 +783,66 @@ pcmk_numeric_strcasecmp(const char *s1, const char *s2)
     }
     return 0;
 }
+
+int
+pcmk__strcmp(const char *s1, const char *s2, uint32_t flags)
+{
+    /* If this flag is set, the second string is a regex. */
+    if (is_set(flags, pcmk__str_regex)) {
+        regex_t *r_patt = calloc(1, sizeof(regex_t));
+        int reg_flags = REG_EXTENDED | (is_set(flags, pcmk__str_casei) ? REG_ICASE : 0);
+        int regcomp_rc = 0;
+        int rc = 0;
+        int nregs = 0;
+        regmatch_t *pmatch = NULL;
+
+        regcomp_rc = regcomp(r_patt, s2, reg_flags);
+        if (regcomp_rc != 0) {
+            rc = 1;
+            crm_err("Bad regex '%s' for update: %s", s2, strerror(regcomp_rc));
+        } else {
+            rc = regexec(r_patt, s1, nregs, pmatch, 0);
+        }
```

Comment:
I think we need a NULL check before this point.

Reply:
Fixed, and added test cases.